### PR TITLE
Feat: Support client no touch in ClientOptions struct

### LINKED ISSUE #236: Support CLIENT NO-EVICT/NO-TOUCH/SETINFO
As titled. These three commands are introduced in Redis 7.2 and needed to be supported in the `rueidis.ClientOption` struct.

### PR DESCRIPTION
**Pull request by SoulPancake**
- Resolves #236

### CODE REVIEW

File: rueidis.go [hunk 1]
```diff
@@ -127,6 +127,8 @@ type ClientOption struct {
 
 	// ShuffleInit is a handy flag that shuffles the InitAddress after passing to the NewClient() if it is true
 	ShuffleInit bool
+	// NoTouch controls whether commands alter LRU/LFU stats
+	NoTouch bool
```
Comment (rueian) [inline rueidis.go:131]: Hi @SoulPancake, Thanks!

I think `ClientNoTouch` would be better, aligning with the current `ClientName` option.

File: rueidis.go [hunk 2]
```diff
@@ -267,6 +269,11 @@ func NewClient(option ClientOption) (client Client, err error) {
 	if option.Dialer.KeepAlive == 0 {
 		option.Dialer.KeepAlive = DefaultTCPKeepAlive
 	}
+	// Check if NoTouch is enabled
+	if option.NoTouch {
+		option.PipelineMultiplex = singleClientMultiplex(option.PipelineMultiplex)
+		return newSingleClient(&option, nil, makeConn)
+	}
```
Comment (rueian) [inline rueidis.go:276]: These lines look wired to me. I am much expecting to modify here: https://github.com/redis/rueidis/blob/main/pipe.go#L133 and here: https://github.com/redis/rueidis/blob/main/pipe.go#L194

**Comment by SoulPancake** [comment]
Does this look fine now? @rueian 
Am I missing something, Apologies as I'm new to this repo and still studying ^_^

File: pipe.go [hunk 3]
```diff
@@ -131,6 +131,11 @@ func _newPipe(connFn func() (net.Conn, error), option *ClientOption, r2ps bool)
 	if option.SelectDB != 0 {
 		init = append(init, []string{"SELECT", strconv.Itoa(option.SelectDB)})
 	}
+	if option.ClientNoTouch {
+		init = append(init, []string{"CLIENT", "NO-TOUCH", "ON"})
+	} else {
+		init = append(init, []string{"CLIENT", "NO-TOUCH", "OFF"})
```
Comment (rueian) [inline pipe.go:137]: It is much better now. However, there is no need to send NO-TOUCH OFF.

Comment (SoulPancake) [reply pipe.go]: Done @rueian

**Comment by rueian** [comment]
Hi @SoulPancake, I think it is almost done. Just need to test them in these two tests: https://github.com/redis/rueidis/blob/main/pipe_test.go#L182-L239

File: pipe_test.go [hunk 4]
```diff
@@ -222,11 +223,12 @@ func TestNewPipe(t *testing.T) {
...
-			Password:     "pa",
-			ClientName:   "cn",
-			AlwaysRESP2:  true,
-			DisableCache: true,
+			SelectDB:      1,
+			Password:      "pa",
+			ClientName:    "cn",
+			AlwaysRESP2:   true,
+			DisableCache:  true,
+			ClientNoTouch: true,
```
Comment (rueian) [inline pipe_test.go:231]: You also need to update the mock expectation above 🤗

File: pipe_test.go [hunk 5]
```diff
@@ -219,6 +221,8 @@ func TestNewPipe(t *testing.T) {
 				ReplyString("OK")
 			mock.Expect("CLIENT", "SETNAME", "cn").
 				ReplyString("OK")
+			mock.Expect("CLIENT", "NO-TOUCH", "ON").
+				ReplyString("OK")
```
Comment (rueian) [inline pipe_test.go:225]: I think you should add it after the SELECT 1

**Pull request merged**